glidertools/plot.py

```diff
@@ -350,16 +353,30 @@ def bin_size(depth, bins=None, ax=None, add_colorbar=True, **hist_kwargs):
         if bins is None:
             binning_freq = 50
             ybins = get_optimal_bins(depth, binning_freq)[0]
+            xbins = r_[nan, diff(ybins)]
+
         else:
-            ybins = bins
-        xbins = r_[nan, diff(ybins)]
+            ybins = bins[1]
+            xbins = []
```

Comment:
The same might be achieved using 

```python
# xbins is equivalent to count and 
# bin_edges is the same as ybins
xbins, bin_edges = np.histogram(y, ybins)  

```

Reply:
I don't think I follow. xbins is not the count, is it? When the bins are defined by the user, xbins gives the median value of the vertical sampling. I chose to do this because the whole point of defining xbins here is to plot the line that gives a good average representation of the sampling rate at the particular depth level. When the bins are optimally calculated (not user specified), defining xbins= diff(ybins) effectively produces that.